### PR TITLE
Add instance node type to associate modal.

### PR DESCRIPTION
**Pull request by kialam**
##### SUMMARY

Related #10853
Related #10830

This PR adds the instance node type to the associate modal:

Screen Shot 2021-09-02 at 5 33 34 PM

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - UI

##### AWX VERSION

```
19.3.0
```

### CODE REVIEW

**Review by AlexSCorey — APPROVED** [review]
I like that change to the Options List.  I think it'll make it easier to convert the rest of the modals that show lists to tables in the future.  Well done!

File: awx/ui/src/components/AssociateModal/AssociateModal.js [hunk 1]
```diff
@@ -18,6 +18,10 @@ const QS_CONFIG = (order_by = 'name') =>
 
 function AssociateModal({
   header = t`Items`,
+  columns = [
+    { key: 'hostname', name: 'Name' },
```
Comment (AlexSCorey) [inline awx/ui/src/components/AssociateModal/AssociateModal.js]:
```suggestion
    { key: 'hostname', name: t`Name` },
```

File: awx/ui/src/components/AssociateModal/AssociateModal.js [hunk 2]
```diff
@@ -18,6 +18,10 @@ const QS_CONFIG = (order_by = 'name') =>
...
+    { key: 'node_type', name: 'Node Type' },
```
Comment (AlexSCorey) [inline awx/ui/src/components/AssociateModal/AssociateModal.js]:
```suggestion
    { key: 'node_type', name:t`Node Type`},
```

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
The translation of the column headers should occur in the AssociateModal.js

File: awx/ui/src/components/AssociateModal/AssociateModal.js (hunk 1 above)
Comment (kialam) [reply awx/ui/src/components/AssociateModal/AssociateModal.js]: @AlexSCorey the translation happens the `buildHeaderRow` function: https://github.com/ansible/awx/pull/11014/files#diff-2e5d5340c567d68ff22083d5a9841eadbb23f76978f860187e77419b079892d7R55

**Review by AlexSCorey — APPROVED** [review]
This looks good.

File: awx/ui/src/components/AssociateModal/AssociateModal.js (hunk 1 above)
Comment (kialam) [reply awx/ui/src/components/AssociateModal/AssociateModal.js]: ^ it appears that dynamic translation of vars doesn't work as I expected it to, so good catch!

**Review by jbradberry — APPROVED** [review]

**Review by akus062381 — APPROVED** [review]

**Pull request merged**